product_pack/models/product_pricelist.py

```diff
@@ -0,0 +1,51 @@
+from odoo import models
+
+
+class Pricelist(models.Model):
+    _inherit = "product.pricelist"
+
+    def _get_product_price(self, product, quantity, uom=None, date=False, **kwargs):
+        """Compute the pricelist price for the specified pack product, qty & uom.
+
+        :returns: unit price of the pack product + components,
+                  considering pricelist rules
+        """
+        self and self.ensure_one()
```

Comment:
Hello @bruno-zanotti , Thanks for the PR.

Is it because you're not sure that self exist that you make self and self.ensure_one()?

Reply:
Yes, this is to be sure that self exists and is at most one record. I don't remember the exact case but at some point odoo run the method without any record and it raises an error. If you check the original method Odoo does the same.